### PR TITLE
Return error when trying to close files with pending operations

### LINKED ISSUE #6357: Misshitsu no Sacrifice and Macross series needs mipmap interpolation for smooth animation.
DEMO: NPJH90092 PRODUCT: ULJS00251
(This game required 6.10 PSP update)

**No vocal**
All vocals are silent in game as reported in #2436 but BGM now works fine.

In debug log:

```
01:24:128 BGM thread   E[ME]: HLE\sceAtrac.cpp:939 sceAtracGetSecondBufferInfo(0, 0bfafa54, 0bfafa58)
01:24:128 BGM thread   D[ME]: HLE\sceAtrac.cpp:1353 sceAtracSetLoopNum(0, -1)
01:24:128 BGM thread   D[ME]: HLE\sceAtrac.cpp:919 sceAtracGetRemainFrame(0, 0bfafa50)
01:24:128 BGM thread   D[IO]: HLE\sceUmd.cpp:316 0x32=sceUmdGetDriveStat()
01:24:129 BGM thread   E[ME]: HLE\sceAtrac.cpp:632 WARNING: Remaining data in packet - we currently only decode one frame/packet
01:24:129 BGM thread   D[ME]: HLE\sceAtrac.cpp:702 00000000=sceAtracDecodeData(0, 08986a00, 0bfafa24[00000800], 0bfafaf8[00000000], 0bfafa1c[232])
```

**Intro background video lagging**

(Check next comment for animation)

Nothing wrong in log console, OP movie plays pretty fine and smooth though.

Edit:
Just a list of titles that are affected by this with gameID's in case it'll ever need a hack:
 - Misshitsu no Sacrifice(ULJS-00251, NPJH-50224),
 - Misshitsu no Sacrifice Jitka: Aru Heisa Shisetsu kara no Dasshutsu (NPJH-00065),
 - Misshitsu no Sacrifice: Miki High Tension Night (NPJH-00077),
 - Macross Ace Frontier (ULJS-00158, ULJS-19029),
 - Macross Ultimate Frontier (ULJS-00231, ULJS-00223),
 - Macross Triangle Frontier (ULJS-00321, ULJS-00369, ULJS-19072).

 Macross series uses it for(AFAIK) water animation only, Misshitsu series uses it for pretty much all animations, including animated UI.

Also to note since things are easily forgotten, mipmap interpolation is already supported in software renderer.

### PR DESCRIPTION
**Pull request by hrydgard**
Also, fix error for trying to close stdin/stderr/stdout.

Blindly tries to help #6357 by trying to do the same thing as https://github.com/jpcsp/jpcsp/commit/61ecddc24341a39667e7d869934ef7ef583fcd03 .

Does it work?

### CODE REVIEW

**Comment by hrydgard** [comment]
That vulkan commit I moved to master. Don't know why it still shows here as a diff..

File: Core/HLE/sceIo.cpp [hunk 1]
```diff
@@ -66,7 +66,8 @@ static const int ERROR_ERRNO_FILE_ALREADY_EXISTS          = 0x80010011;
 static const int ERROR_MEMSTICK_DEVCTL_BAD_PARAMS         = 0x80220081;
 static const int ERROR_MEMSTICK_DEVCTL_TOO_MANY_CALLBACKS = 0x80220082;
 static const int ERROR_KERNEL_BAD_FILE_DESCRIPTOR         = 0x80020323;
-
+static const int ERROR_KERNEL_ILLEGAL_PERMISSION          = 0x800200d1;
```
Comment (unknownbrackets) [inline Core/HLE/sceIo.cpp]: Should we use SCE_KERNEL_ERROR_ILLEGAL_PERM instead?  Same value, and used other places.

-[Unknown]

File: Core/HLE/sceIo.cpp [hunk 2]
```diff
@@ -66,7 +66,8 @@ static const int ERROR_ERRNO_FILE_ALREADY_EXISTS          = 0x80010011;
...
+static const int ERROR_KERNEL_ASYNC_BUSY                  = 0x80020329;
```
Comment (unknownbrackets) [inline Core/HLE/sceIo.cpp]: We use `SCE_KERNEL_ERROR_ASYNC_BUSY` already in this file for this error code.

-[Unknown]

**Comment by unknownbrackets** [comment]
This does fix it in the trial, yes.  It doesn't fix the animation "tweening" by mip bias, though.

-[Unknown]

File: Core/HLE/sceIo.cpp (hunk 2 above)
Comment (hrydgard) [inline Core/HLE/sceIo.cpp]: Oops, thanks.

File: Core/HLE/sceIo.cpp (hunk 1 above)
Comment (hrydgard) [inline Core/HLE/sceIo.cpp]: I suppose so. I searched the code base for ILLEGAL_PERMISSION, didn't anticipate just "PERM"..

**Comment by hrydgard** [comment]
Well, yeah, it's obviously not expected to fix the mipmap trickery, that should really be a separate issue.

File: Core/HLE/sceIo.cpp (hunk 1 above)
Comment (unknownbrackets) [inline Core/HLE/sceIo.cpp]: Yeah.... most of those names came from the "snapshot" commit, not sure.

-[Unknown]

**Comment by unknownbrackets** [comment]
Yeah, of course, was just noting it for people who don't read the code...

-[Unknown]

**Comment by sum2012** [comment]
Should add back ERROR_KERNEL_BAD_FILE_DESCRIPTOR error code
https://github.com/jpcsp/jpcsp/commit/31f4f078cd7d367bcf84fbc7a9a4c73125d82069

**Comment by hrydgard** [comment]
@sum2012 we already return SCE_KERNEL_ERROR_BADF in that case, see https://github.com/hrydgard/ppsspp/blob/master/Core/HLE/sceIo.cpp#L1902

**Comment by sum2012** [comment]
@hrydgard I see
https://github.com/hrydgard/ppsspp/blob/master/Core/HLE/sceKernel.h#L292

**Pull request merged**